### PR TITLE
build: fix static builds

### PR DESCRIPTION
**Pull request by bassamtabbara**
restructure the build tags and installsuffix usage for golang
to build our projects correctly for static and dynamic builds.

### CODE REVIEW

File: build/cross-image/Dockerfile [hunk 1]
```diff
@@ -161,15 +161,24 @@ RUN GO_VERSION=1.7.3 && \
 ENV PATH /usr/local/go/bin:$PATH
 
 # precompile the go standard library for all supported platforms and configurations
+# there are three configurations that we compile for:
```
Comment (jbw976) [inline build/cross-image/Dockerfile]: does this change the typical dev flow invocation on our build?  we've generally been using `build/run make -j4 VERSION=dev CLIENT_SERVER_PLATFORMS=linux_amd64 release`, does that change at all?

**Comment by DoubleDensity** [comment]
OK to test

**Comment by DoubleDensity** [comment]
OK to test

**Comment by DoubleDensity** [comment]
OK to test

**Pull request merged**